### PR TITLE
[4/n] Implements payment flow

### PR DESCRIPTION
**Pull request by e-wai**
## Notion ticket link

[Registrant Experience: Payment Failure/Confirmation](https://www.notion.so/uwblueprintexecs/Task-Board-F22-9e03bf9f7c4343f09a8a5c16dc48a12a?p=c4a45ddbdb1f433cbf01448ede701316&pm=c)
[Payments - Stripe API](https://www.notion.so/uwblueprintexecs/Task-Board-F22-9e03bf9f7c4343f09a8a5c16dc48a12a?p=5758698f7e8d440da53b92b84082df3c&pm=c)

## Implementation description
* Used session storage to cache session details and restore session when the user clicks "back"
  * `sessionStorage` is different from `localStorage`, expires: https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage 
* Creates a bunch of components for result handling in `RegistrationResult/`

Based on work in https://github.com/uwblueprint/focus-on-nature/pull/117:
* To get around CORS errors for redirect, let's return the created checkout link and then initiate the movement to the stripe page in the frontend
* Tried adding this to the CORS allowlist but it wasn't enough, I suspect it's bc we can't control the Stripe origin?
* Also adds default camp image

The full e2e flow looks like:
* User goes through registration flow, clicks register
* Request sent to backend to create campers in db, and create checkout session in Stripe
  * If success, return the checkout session URL
  * If failed, return to checkout and show the checkout error modal
* Save session data (see `CheckoutData`) to session storage, including checkout URL, and redirect to checkout
  * note: handling for pressing back on webpage isn't great, is it worth using `window.location.replace()` instead of `window.location.assign()`?
* Stripe checkout
  * If payment success, Stripe redirects to `/register/camp/:campId?result=success`. We retrieve session storage for this page to load summary; if this fails, we have a generic message that says check email
  * If payment fails, Stripe handles that in the webpage (no redirect)
  * If payment cancelled (pressed back arrow on the Stripe checkout page), Stripe redirects to `/register/camp/:campId?result=cancel`. We restore session details from session storage and show error modal.

## Steps to test
1. Go through all checkout/payment flows using a Stripe test card (https://stripe.com/docs/testing)
  a. cancel (press back button) 
  b. complete payment -> should end up at success page
  c. incorrect card details / payment decline (see Stripe docs)
2. Look for success page states: through Stripe payment, also access directly and look at error state
3. Check mobile states for all of above (I think the modal is too wide on phone?)
4. Mess around with session Storage (invalidate it manually, see if app handles it)

## What should reviewers focus on?
* Look at `frontend/` only please -- backend created in separate PR https://github.com/uwblueprint/focus-on-nature/pull/212, and will rebase later

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/RegistrantExperience/RegistrationResult/index.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+const PaymentSummaryList = ({
+  items,
+}: {
+  items: CartItem[];
+}): React.ReactElement => {
+  return (
+    <VStack w={{ base: "100%", lg: "35vw" }} spacing={10}>
+      <VStack w="100%" overflowY="auto" h={{ lg: "35vh" }} spacing={8}>
+        {/* TODO switch to sum of EDLP, non-EDLP items */}
```
Comment (dhmi19) [inline frontend/src/components/pages/RegistrantExperience/RegistrationResult/index.tsx:45]: not sure what this is referring to exactly but did you want to address it in this PR?

File: frontend/src/components/pages/RegistrantExperience/RegistrationSteps/RegistrationFooter.tsx [hunk 2]
```diff
@@ -6,21 +6,23 @@ export type RegistrationFooterProps = {
...
+  registrationLoading,
   handleStepNavigation,
 }: RegistrationFooterProps): React.ReactElement => {
   const onNextStep = () => {
     if (isCurrentStepCompleted) {
       handleStepNavigation(1);
     } else {
       alert(
-        "Form does not pass validaiton. Please complete all form fields according to requirements.",
+        "Form does not pass validation. Please complete all form fields according to requirements.",
```
Comment (dhmi19) [inline frontend/src/components/pages/RegistrantExperience/RegistrationSteps/RegistrationFooter.tsx:25]: I was actually hoping to remove the alert. I don't think the alerts are standard practise and the error states of the fields does the same thing

**Review by dhmi19 — APPROVED** [review]
you are a legend

File: frontend/src/components/pages/RegistrantExperience/RegistrationSteps/RegistrationFooter.tsx (hunk 2 above)
Comment (e-wai) [reply frontend/src/components/pages/RegistrantExperience/RegistrationSteps/RegistrationFooter.tsx:25]: How about a toast?

File: frontend/src/components/pages/RegistrantExperience/RegistrationResult/index.tsx (hunk 1 above)
Comment (e-wai) [reply frontend/src/components/pages/RegistrantExperience/RegistrationResult/index.tsx:45]: nah, I handle it in the EDLP fixes PR

**Pull request merged**